### PR TITLE
bug(dsg 2302): Schema builder element improvements

### PR DESCRIPTION
**Pull request by thechechogomez**
# Basics

## PR Type

What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

Ticket: [DSG-2302](https://dol-ewf.atlassian.net/browse/DSG-2302)

It fixes the camel case format on the schema builder component name. We can now write `nameA` without throwing an error. 

It removes the possibility of adding the List component as the `content type` for the List Component. 

It removes the possibility of adding the currently editing schema to the `schema type` on the Schema Component.

## Screenshot of the affected changes

Camel case fix on component name
https://github.com/Nuvalence/dsgov-web/assets/60438413/1beb47fa-3902-4a68-8b0b-98c26e37e0a9

Don't display List as an available option for the List component
https://github.com/Nuvalence/dsgov-web/assets/60438413/db07ef53-66e8-4c2f-a55b-9e0c07d52170

Don't display current schema as an available option in Schema Component
https://github.com/Nuvalence/dsgov-web/assets/60438413/3f951111-2ec7-472f-9741-d6e6673773a3

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

[DSG-2302]: https://dol-ewf.atlassian.net/browse/DSG-2302?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/shared/feature/schema-nuv/src/lib/components/schema-select/formio/formio-schema-select.component.ts [hunk 1]
```diff
@@ -25,28 +26,39 @@ export class FormioSchemaSelectComponent extends FormioBaseCustomComponent<ISche
   public formControl = new FormControl();
   public gotSchemas = false;
 
-  constructor(private readonly _workApiRoutesService: WorkApiRoutesService, private readonly _changeDetectorRef: ChangeDetectorRef) {
+  private schemaKey = '';
+  private readonly getSchemaKey$ = this._schemaBuilderService.metaDataFields$.pipe(tap(data => (this.schemaKey = data.key)));
```
Comment (thechechogomez) [inline libs/shared/feature/schema-nuv/src/lib/components/schema-select/formio/formio-schema-select.component.ts:30]: This is a workaround to the fact that using `router.paramMap` doesn't return the true value for the `schemaKey` param.

File: libs/shared/feature/schema-nuv/src/lib/components/base/formio/formio-builder-display.model.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ export const defaultPropertyNameConfiguration = {
   tooltip: `Enter text in camel case format. For example: 'camelCaseText'`,
   type: 'textfield',
   validate: {
-    custom: 'valid = /^[a-z]+([A-Z][a-z0-9]+)*$/.test(input);',
+    custom: 'valid = /^[a-z]+[A-Z0-9a-z]+$/.test(input);',
```
Comment (davidyoung77) [inline libs/shared/feature/schema-nuv/src/lib/components/base/formio/formio-builder-display.model.ts:10]: Can we add tests to verify this fix?

**Review by davidyoung77 — COMMENTED** [review]
These changes look good, please add tests for these updates

File: libs/shared/feature/schema-nuv/src/lib/components/schema-select/formio/formio-schema-select.component.ts [hunk 3]
```diff
@@ -25,28 +26,39 @@ export class FormioSchemaSelectComponent extends FormioBaseCustomComponent<ISche
...
+  constructor(
+    private readonly _workApiRoutesService: WorkApiRoutesService,
+    private readonly _changeDetectorRef: ChangeDetectorRef,
+    private readonly _schemaBuilderService: SchemaBuilderService,
+  ) {
     super();
   }
 
   public ngOnInit(): void {
+    this.getSchemaKey$.subscribe();
```
Comment (davidyoung77) [inline libs/shared/feature/schema-nuv/src/lib/components/schema-select/formio/formio-schema-select.component.ts:41]: Lets prevent the memory leak
```suggestion
    this.getSchemaKey$.pipe(take(1)).subscribe();
```

**Review by davidyoung77 — APPROVED** [review]
Approved pending comments

**Review by jgnuvalence — APPROVED** [review]

**Pull request merged**